namesys/namesys.go

```diff
@@ -87,9 +89,44 @@ func (ns *mpns) resolveOnce(ctx context.Context, name string) (path.Path, error)
 
 // Publish implements Publisher
 func (ns *mpns) Publish(ctx context.Context, name ci.PrivKey, value path.Path) error {
-	return ns.publishers["/ipns/"].Publish(ctx, name, value)
+	err := ns.publishers["/ipns/"].Publish(ctx, name, value)
+	if err != nil {
+		return err
+	}
+	ns.addToDHTCache(name, value, time.Now().Add(time.Hour*24))
```

Comment:
Lets just call it `DefaultRecordTTL` (i had to go look at it again to remember what exactly this was)